Evaluate-ModelarDB-Compression/main.py

```diff
@@ -0,0 +1,281 @@
+
+
+def errors_occurred(output_stream):
+    normalized = output_stream.lower()
+    return b"error" in normalized or b"panicked" in normalized
+
+
+def print_stream(output_stream):
+    print()
+    print(output_stream.decode("utf-8"))
+
+
+def ingest_test_data(utilities_loader, test_data, error_bound):
+    process = subprocess.run(
+        [
+            "python3",
+            utilities_loader,
+            "127.0.0.1:9999",
+            TABLE_NAME,
+            test_data,
+            error_bound,
+        ],
+        stdout=STDOUT,
+        stderr=STDERR,
+    )
+
+    if errors_occurred(process.stderr):
+        print_stream(process.stderr)
+        return True
+
+
+def retrieve_schema(flight_client):
+    flight_descriptor = flight.FlightDescriptor.for_path(TABLE_NAME)
+    schema_result = flight_client.get_schema(flight_descriptor)
+    return schema_result.schema
+
+
+def retrieve_ingested_column(flight_client, column_name, timestamp_column):
+    ticket = flight.Ticket(
+        "SELECT "
+        + column_name
+        + " FROM "
+        + TABLE_NAME
+        + " ORDER BY "
+        + timestamp_column
+    )
+    reader = flight_client.do_get(ticket)
+    return reader.read_all().column(column_name)
+
+
+def compute_and_print_metrics(real_column, compressed_column, error_bound):
+    # Arrays make iteration simpler and float32 match ModelarDB's precision.
+    real_column = real_column.to_numpy().astype(numpy.float32)
+    compressed_column = compressed_column.to_numpy()
+
+    # Initialize variables for computing metrics.
+    equal_values = 0
+    compared_values = 0
+
+    sum_difference = 0.0
+    sum_real_values = 0.0
+
+    max_actual_error = 0.0
+    max_actual_error_real_value = 0.0
+    max_actual_error_compressed_value = 0.0
+
+    ceiled_actual_errors = math.ceil(error_bound + 2) * [0]
+
+    # Compute metrics.
+    for real_value, compressed_value in zip(real_column, compressed_column):
+        if real_value == compressed_value:
+            equal_values += 1.0
+            difference = 0.0
+            actual_error = 0.0
+        else:
+            difference = real_value - compressed_value
+            actual_error = abs(difference / real_value)
+
+        compared_values += 1
+
+        sum_difference += difference
+        sum_real_values += real_value
+
+        if max_actual_error < actual_error:
+            max_actual_error = actual_error
+            max_actual_error_real_value = real_value
+            max_actual_error_compressed_value = compressed_value
+
+        try:
+            ceiled_actual_errors[math.ceil(actual_error)] += 1
+        except OverflowError:
+            print(
+                "ERROR: undefined error due to {} (real) and {} (compressed).".format(
+                    real_value, compressed_value
+                )
+            )
+            print()
+            return
+
+    # Compute and print the final result.
+    print("- Real Values: {}".format(len(real_column)))
```

Comment:
Maybe use [f-strings](https://peps.python.org/pep-0498/) instead to make it a little more readable.

Reply:
Fixed.